### DISCUSSION TITLE
Netmiko Palo alto fails when sending commit using send_config_set

### DISCUSSION THREAD

**Discussion by akshayavk96** [ktbyers/netmiko#2741] [Q&A] [answered]
Hi everyone,
I am using Netmiko for Palo alto to do some automation. When I send commit command using send_config_set , it throws me a read timeout exception. 
    output += self.read_until_pattern(pattern=pattern, re_flags=re.M)
 in read_until_pattern
    raise ReadTimeout(msg)
netmiko.exceptions.ReadTimeout: 
Pattern not detected:
Things you might try to fix this:
1. Adjust the regex pattern to better identify the terminating string. Note, in
many situations the pattern is automatically based on the network device's prompt.
2. Increase the read_timeout to a larger value.
I tried increasing read_timeout to a larger value, but the output remains the same. Could someone please help me with this issue? Any help would be appreciated.

**Comment by ktbyers** [comment, marked as answer]
@akshayavk96 You should use the `commit` method for this.

```python
    def commit(
        self,
        comment: str = "", 
        force: bool = False,
        partial: bool = False,
        device_and_network: bool = False,
        policy_and_objects: bool = False,
        vsys: str = "", 
        no_vsys: bool = False,
        read_timeout: float = 120.0,
        delay_factor: Optional[float] = None,
    ) -> str:
```

**Reply by akshayavk96** [reply]
Thank you @ktbyers
